sdk/textanalytics/Azure.AI.TextAnalytics/src/Transforms.Legacy.cs

```diff
@@ -0,0 +1,641 @@
+            {
+                Legacy.Models.DocumentSentimentValue.Neutral => TextSentiment.Neutral,
+                Legacy.Models.DocumentSentimentValue.Positive => TextSentiment.Positive,
+                Legacy.Models.DocumentSentimentValue.Negative => TextSentiment.Negative,
+                _ => throw new NotSupportedException($"The document sentiment, { sentiment }, is not supported for conversion.")
+            };
+
+        //TODO:  Check this conversion.  Seems to match: https://github.com/Azure/azure-sdk-for-net/blob/Azure.AI.TextAnalytics_5.2.0-beta.2/sdk/textanalytics/Azure.AI.TextAnalytics/src/DocumentSentiment.cs#L56
+        internal static List<SentenceOpinion> ConvertToSentenceOpinions(IReadOnlyList<Legacy.SentenceAssessment> assessments)
+        {
+            var assessmentList = new List<AssessmentSentiment>(assessments.Count);
+            var targetList = new List<TargetSentiment>(assessments.Count);
+
+            foreach (var assessment in assessments)
+            {
+                targetList.Add(new TargetSentiment(
+                    ConvertToTextSentiment(assessment.Sentiment),
+                    assessment.Text,
+                    assessment.ConfidenceScores.Positive,
+                    assessment.ConfidenceScores.Negative,
+                    assessment.Offset,
+                    assessment.Length));
+
+                assessmentList.Add(new AssessmentSentiment(
+                    ConvertToTextSentiment(assessment.Sentiment),
+                    assessment.ConfidenceScores.Positive,
+                    assessment.ConfidenceScores.Negative,
+                    assessment.Text,
+                    assessment.IsNegated,
+                    assessment.Offset,
+                    assessment.Length));
+            }
+
+            var opinions = new List<SentenceOpinion>(targetList.Count);
+
+            foreach (var target in targetList)
+            {
+                opinions.Add(new SentenceOpinion(target, assessmentList));
+            }
+
+            return opinions;
+        }
+
+        //TODO: Check this conversion.  The assessment conversion is odd, but seems to match the legacy version.
+        internal static List<SentenceSentiment> ConvertToSentenceSentiments(IReadOnlyList<Legacy.SentenceSentiment> legacySentiments)
+        {
+            var sentences = new List<SentenceSentiment>(legacySentiments.Count);
+
+            foreach (var sentiment in legacySentiments)
+            {
+                sentences.Add(new SentenceSentiment(
+                    ConvertToTextSentiment(sentiment.Sentiment),
+                    sentiment.Text,
+                    sentiment.ConfidenceScores.Positive,
+                    sentiment.ConfidenceScores.Neutral,
+                    sentiment.ConfidenceScores.Negative,
+                    sentiment.Offset,
+                    sentiment.Length,
+                    ConvertToSentenceOpinions(sentiment.Assessments)));
+            }
```

Comment:
nit: would be easier to read 
```suggestion
            foreach (var sentence in legacySentiments)
            {
                sentences.Add(new SentenceSentiment(
                    ConvertToTextSentiment(sentence.Sentiment),
                    sentence.Text,
                    sentence.ConfidenceScores.Positive,
                    sentence.ConfidenceScores.Neutral,
                    sentence.ConfidenceScores.Negative,
                    sentence.Offset,
                    sentence.Length,
                    ConvertToSentenceOpinions(sentence.Assessments)));
            }
```

Reply:
Changed all `sentiment` references in the method to `sentence`.